File: 2017 Solutions/Freshman/Day1.py
# Part 1

userInput = input()
sum = 0
for i in range(len(userInput) - 1):
    if userInput[i] == userInput[i+1]:
        sum += int(userInput[i])
if userInput[len(userInput) - 1] == userInput[0]:
    sum += int(userInput[len(userInput) - 1])

print(sum)

# Part 2

sum = 0
for i in range(len(userInput)):
    if userInput[i] == userInput[(i + len(userInput) // 2) % len(userInput)]:
        sum += int(userInput[i])

print(sum)

File: 2019 Solutions/Junior/day1.cpp
#include <iostream>

using namespace std;

int main() {
    freopen("input.in", "r", stdin);

//    Part 1
//    int ans = 0;
//
//    while (!cin.eof()) {
//        int n;
//        cin >> n;
//        ans += (n / 3 - 2);
//    }
//
//    cout << ans << endl;


//    Part 2
    int ans = 0;

    while (!cin.eof()) {
        int n;
        cin >> n;
        n = n / 3 - 2;
        while (n > 0) {
            ans += n;
            n = n / 3 - 2;
        }
    }

    cout << ans << endl;
    return 0;
}